### PR TITLE
0.4.2

### PR DESCRIPTION
**Pull request by jmonteer**

### CODE REVIEW

File: contracts/AaveLenderBorrowerCloner.sol [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    event Deployed(address indexed original);
+
+    constructor(
+        address _vault,
+        address _yVault,
+        bool _isWantIncentivised,
+        bool _isInvestmentTokenIncentivised,
+        string memory _strategyName
+    ) public {
+        Strategy _original = new Strategy(_vault, _yVault, _strategyName);
```
Comment (poolpitako) [inline contracts/AaveLenderBorrowerCloner.sol:20]: I would do `original = xxx` directly.

File: contracts/AaveLenderBorrowerCloner.sol [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+            1 // maxLoss (default: 1)
+        );
+
+        Strategy(_original).setRewards(msg.sender);
+        Strategy(_original).setKeeper(msg.sender);
+        Strategy(_original).setStrategist(msg.sender);
+    }
+
+    function name() external pure returns (string memory) {
+        return "Yearn-AaveLenderBorrowerCloner";
```
Comment (poolpitako) [inline contracts/AaveLenderBorrowerCloner.sol:42]: `Yearn-AaveLenderBorrowerCloner@0.4.2-1` or perhaps use `original.apiVersion()`

File: contracts/libraries/AaveLenderBorrowerLib.sol [hunk 3]
```diff
@@ -0,0 +1,285 @@
...
+    function protocolDataProvider()
+        public
+        view
+        returns (IProtocolDataProvider)
+    {
+        return
+            IProtocolDataProvider(0x057835Ad21a177dbdd3090bB1CAE03EaCF78Fc6d);
+    }
+
+    function MAX_BPS() internal view returns (uint256) {
```
Comment (poolpitako) [inline contracts/libraries/AaveLenderBorrowerLib.sol:48]: uff you can't define constants in a library? making it pure might reduce contract size.

File: contracts/Strategy.sol [hunk 4]
```diff
@@ -73,8 +72,8 @@ contract Strategy is BaseStrategy {
     IStakedAave internal constant stkAave =
         IStakedAave(0x4da27a545c0c5B758a6BA100e3a049001de870f5);
 
-    IProtocolDataProvider internal constant protocolDataProvider =
-        IProtocolDataProvider(0x057835Ad21a177dbdd3090bB1CAE03EaCF78Fc6d);
+    // IProtocolDataProvider internal constant protocolDataProvider =
+    //     IProtocolDataProvider(0x057835Ad21a177dbdd3090bB1CAE03EaCF78Fc6d);
```
Comment (poolpitako) [inline contracts/Strategy.sol:76]: remove

File: contracts/AaveLenderBorrowerCloner.sol [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+    address public immutable original;
+
+    event Cloned(address indexed clone);
+    event Deployed(address indexed original);
+
+    constructor(
+        address _vault,
+        address _yVault,
+        bool _isWantIncentivised,
+        bool _isInvestmentTokenIncentivised,
```
Comment (poolpitako) [inline contracts/AaveLenderBorrowerCloner.sol:17]: it's weird that this contructor doesn't match the one in the strategy. AFAIK there is a second call which would `setStrategyParams`  after a clone, so I would get rid of those two bools here and default is false.

File: contracts/AaveLenderBorrowerCloner.sol [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+        }
+
+        Strategy(newStrategy).initialize(_vault, _yVault, _strategyName);
+
+        Strategy(newStrategy).setStrategyParams(
+            4_000, // targetLTVMultiplier (default: 4_000)
+            6_000, // warningLTVMultiplier default: 6_000
+            1e27, // acceptableCosts (default: 1e27)
+            7, // default: 7 (Yearn Aave Referal code)
+            type(uint256).max, // 2**256
```
Comment (poolpitako) [inline contracts/AaveLenderBorrowerCloner.sol:79]: haha comment should be `// max debt to take`

File: contracts/Strategy.sol [hunk 7]
```diff
@@ -673,13 +603,11 @@ contract Strategy is BaseStrategy {
 
     function _calculateAmountToRepay(uint256 amount)
         internal
-        view
```
Comment (poolpitako) [inline contracts/Strategy.sol:676]: why?

Comment (jmonteer) [reply contracts/Strategy.sol:676]: removed then forgot to add back

good catch

File: contracts/AaveLenderBorrowerCloner.sol (hunk 5 above)
Comment (jmonteer) [reply contracts/AaveLenderBorrowerCloner.sol:17]: second call would not be required if we are OK with default params

having those bools incorrectly set might cause problems so I would leave it like it is atm

File: contracts/libraries/AaveLenderBorrowerLib.sol (hunk 3 above)
Comment (jmonteer) [reply contracts/libraries/AaveLenderBorrowerLib.sol:48]: brain fart :D

you can set constants, you cannot set state variables

File: contracts/AaveLenderBorrowerCloner.sol (hunk 1 above)
Comment (jmonteer) [reply contracts/AaveLenderBorrowerCloner.sol:20]: not important but it saves gas to use a local variable vs a state variable

**Comment by jmonteer** [comment]
Current Bytecode Size status

image

**Comment by poolpitako** [comment]
Replaced by https://github.com/jmonteer/yearnV2-aave-lender-borrower/pull/22

**Pull request closed without merging**